src/disco/components/Addon.js

```diff
@@ -1,34 +1,66 @@
 import React, { PropTypes } from 'react';
 import classNames from 'classnames';
+import themeAction from 'disco/themePreview';
+import { gettext as _ } from 'core/utils';
 
 import InstallButton from './InstallButton';
+import {
+  validAddonTypes,
+  EXTENSION_TYPE,
+  THEME_TYPE,
+  THEME_PREVIEW,
+  THEME_RESET_PREVIEW,
+} from 'disco/constants';
 
 import 'disco/css/Addon.scss';
 
 
 export default class Addon extends React.Component {
   static propTypes = {
-    id: PropTypes.number.isRequired,
-    type: PropTypes.string.isRequired,
+    accentcolor: PropTypes.string,
+    editorialDescription: PropTypes.string.isRequired,
+    footerURL: PropTypes.string,
+    headerURL: PropTypes.string,
     heading: PropTypes.string.isRequired,
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
     subHeading: PropTypes.string,
-    editorialDescription: PropTypes.string.isRequired,
+    textcolor: PropTypes.string,
+    type: PropTypes.oneOf(validAddonTypes).isRequired,
+    themeAction: PropTypes.func,
+  }
+
+  static defaultProps = {
+    // Defaults themeAction to the imported func.
+    themeAction,
+  }
+
+  getBrowserThemeData() {
+    const { id, name, headerURL, footerURL, textcolor, accentcolor } = this.props;
+    return JSON.stringify({id, name, headerURL, footerURL, textcolor, accentcolor});
   }
 
   getLogo() {
     const { id } = this.props;
     const imageURL = `https://addons-dev-cdn.allizom.org/user-media/addon_icons/0/${id}-64.png?modified=1388632826`;
-    if (this.props.type === 'Extension') {
+    if (this.props.type === EXTENSION_TYPE) {
       return <div className="logo"><img src={imageURL} alt="" /></div>;
     }
     return null;
   }
 
   getThemeImage() {
-    const { id } = this.props;
+    const { id, name } = this.props;
     const themeURL = `https://addons-dev-cdn.allizom.org/user-media/addons/${id}/preview_large.jpg?1239806327`;
```

Comment:
The `headerURL` and `footerURL` are included in the fake data. Should this go in there too? I don't think we'll be constructing this by hand.